lib/httpoison/base.ex

```diff
@@ -347,16 +382,29 @@ defmodule HTTPoison.Base do
 
       Options: see type `HTTPoison.Request`
 
-      This function returns `{:ok, response}` or `{:ok, async_response}` if the
-      request is successful, `{:error, reason}` otherwise.
+      This function returns `{:ok, response}`, `{:ok, async_response}`, or `{:ok, maybe_redirect}`
+      if the request is successful, `{:error, reason}` otherwise.
+
+      ## Redirect handling
+
+      If the option `:follow_redirect` is given, HTTP redirects are automatically follow if
```

Comment:
Not sure if you'd rather have this information on this function or just add a note to see the documentation for the `MaybeRedirect` struct?

Reply:
I think it's fine like this 👍